internals/search.py

```diff
@@ -125,40 +126,46 @@ def process_queriable_field(
 # A value that a feature field can be compared against.  It can be
 # a single word or a quoted string.
 VALUE_PATTERN = r'[^" ]+|"[^"]+"'
+# Logical operators.
+# TODO(kyleju): support 'OR' logic
+LOGICAL_OPERATORS_PATTERN = r'AND|OR|NOT'
```

Comment:
I should have specified this before:
* For logical OR, the user should enter the keyword `OR`
* Logical AND is implicit, so users don't need to enter that.  If they do enter `AND` it would already be treated as a full-text stop word which return None which does not affect search results.
* For negation, the user should precede the query term with a leading minus sign, e.g.:
    *  `-starred_by:me` 
    * `-owner=user@example.com`
    *  `-word` 
    *  `-"phrase of words"`